tests/backends/base/test_base.py

```diff
@@ -30,3 +32,113 @@ def test_initialization_class_attributes(self):
     def test_initialization_display_name(self):
         self.assertEqual(BaseDatabaseWrapper.display_name, 'unknown')
         self.assertNotEqual(connection.display_name, 'unknown')
+
+
+class ExecuteWrapperTests(TestCase):
+
+    @staticmethod
+    def call_execute(connection, params=None):
+        ret_val = '1' if params is None else '%s'
+        sql = 'SELECT ' + ret_val + connection.features.bare_select_suffix
+        with connection.cursor() as cursor:
+            cursor.execute(sql, params)
+
+    def call_executemany(self, connection, params_list=((1,),) * 3):
+        sql = 'GIBBERISH with SELECT in it %s'
+        # We need a no-op query, but it has to be DML. We could use a model
+        # to make sure there's a table we can update, or we could just catch
+        # the error.
+        with self.assertRaises(Exception):
+            with connection.cursor() as cursor:
```

Comment:
you can collapse, `with self.assertRaises(Exception), connection.cursor() as cursor:`, and in a few places below.

Reply:
Yep.